Error 0x57: The "version" parameter is not supported\MSBuild

With the new version 2.19 of the plugin on our build system we receive:

```
:msbuild FAILED
   Task: :msbuild took 1843ms

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':msbuild'.
> java.io.FileNotFoundException: C:\jenkins\workspace\Build-PDFC-ProcessBridge-NET\PDFCProcessBridge.NET\Visual Studio Locator version 2.3.7+gfb01ddad81
  Copyright (C) Microsoft Corporation. All rights reserved.
  
  Error 0x57: The "version" parameter is not supported\MSBuild
```
.....
```
Caused by: java.io.FileNotFoundException: C:\jenkins\workspace\Build-PDFC-ProcessBridge-NET\PDFCProcessBridge.NET\Visual Studio Locator version 2.3.7+gfb01ddad81
Copyright (C) Microsoft Corporation. All rights reserved.

Error 0x57: The "version" parameter is not supported\MSBuild
	at com.ullink.MsbuildResolver.findMsbuildByVsWhere(MsbuildResolver.groovy:36)
	at com.ullink.MsbuildResolver$findMsbuildByVsWhere.callStatic(Unknown Source)
	at com.ullink.MsbuildResolver.setupExecutable(MsbuildResolver.groovy:62)
	at com.ullink.IExecutableResolver$setupExecutable$0.call(Unknown Source)
	at com.ullink.Msbuild.getCommandLineArgs(Msbuild.groovy:156)
	at com.ullink.Msbuild_Decorated.getCommandLineArgs(Unknown Source)
	at org.gradle.internal.metaobject.BeanDynamicObject$MetaClassAdapter.invokeMethod(BeanDynamicObject.java:479)
	at org.gradle.internal.metaobject.BeanDynamicObject.tryInvokeMethod(BeanDynamicObject.java:191)
	at org.gradle.internal.metaobject.CompositeDynamicObject.tryInvokeMethod(CompositeDynamicObject.java:98)
	at org.gradle.internal.metaobject.MixInClosurePropertiesAsMethodsDynamicObject.tryInvokeMethod(MixInClosurePropertiesAsMethodsDynamicObject.java:30)
	at org.gradle.internal.metaobject.ConfigureDelegate.invokeMethod(ConfigureDelegate.java:80)
	at com.ullink.Msbuild$_build_closure7.doCall(Msbuild.groovy:151)
	at org.gradle.api.internal.ClosureBackedAction.execute(ClosureBackedAction.java:71)
	at org.gradle.util.ConfigureUtil.configureTarget(ConfigureUtil.java:160)
	at org.gradle.util.ConfigureUtil.configure(ConfigureUtil.java:106)
	at org.gradle.util.ConfigureUtil$1.execute(ConfigureUtil.java:123)
	at org.gradle.api.internal.file.DefaultFileOperations.exec(DefaultFileOperations.java:187)
	at org.gradle.api.internal.project.DefaultProject.exec(DefaultProject.java:1076)
	at org.gradle.api.internal.project.DefaultProject.exec(DefaultProject.java:1071)
	at org.gradle.api.Project$exec$4.call(Unknown Source)
	at com.ullink.Msbuild.build(Msbuild.groovy:150)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
	... 41 more
```

--- a/build.gradle
+++ b/build.gradle
@@ -58,7 +58,7 @@ task generateZip(dependsOn: nunit, type: Zip) {
 }
 
 task downloadVsWhere(type: de.undercouch.gradle.tasks.download.Download) {
-    src 'https://github.com/Microsoft/vswhere/releases/download/2.3.7/vswhere.exe'
+    src 'https://github.com/Microsoft/vswhere/releases/download/2.4.1/vswhere.exe'
     dest "$temporaryDir/vswhere.exe"
 }
 

--- a/src/main/groovy/com/ullink/MsbuildResolver.groovy
+++ b/src/main/groovy/com/ullink/MsbuildResolver.groovy
@@ -30,13 +30,17 @@ class MsbuildResolver implements IExecutableResolver {
         def proc = vswhere.start()
         proc.waitFor()
         def location = proc.in.text?.trim();
-        if (location) {
-            def msbuildDir = new File(location, 'MSBuild')
-            msbuild.logger.info("Found following MSBuild installation folder: ${msbuildDir}")
-            msbuildDir.eachDirMatch(~/\d+(\.\d+)*/) { dir ->
-                msbuild.msbuildDir = new File(dir, 'Bin')
-                return
-            }
+        if (!location) {
+            return
+        }
+        def msbuildDir = new File(location, 'MSBuild')
+        if (!msbuildDir.exists()) {
+            return
+        }
+        msbuild.logger.info("Found following MSBuild installation folder: ${msbuildDir}")
+        msbuildDir.eachDirMatch(~/\d+(\.\d+)*/) { dir ->
+            msbuild.msbuildDir = new File(dir, 'Bin')
+            return
         }
     }
 

